Engine/src/main/java/dk/gruppesex/bodtrd/engine/Game.java

```diff
@@ -81,14 +95,66 @@ public void render()
         Gdx.gl.glClear(GL20.GL_COLOR_BUFFER_BIT);
 
         _gameData.setDeltaTime(Gdx.graphics.getDeltaTime());
+
+        update();
+        draw();
+
+    }
+
+    private void loadViews()
+    {
+        for (Entity e : _world.values())
+        {
+            View view = e.get(View.class);
+
+            if (view == null)
+            {
+                continue;
+            }
+            String imagePath = view.getImageFilePath();
+
+            if (!_assetManager.isLoaded(imagePath, Texture.class))
+            {
+                _assetManager.load(imagePath, Texture.class);
+            }
+
+        }
+    }
+
+    private void update()
+    {
         for (IEntityProcessor entityProcessorService : _entityProcessors)
         {
             entityProcessorService.process(_gameData, _world);
         }
+        _assetManager.update();
+    }
 
-        _camera.position.x = (float)_gameData.getPlayerPosition().getX();
-        _camera.position.y = (float)_gameData.getPlayerPosition().getY();
+    private void draw()
+    {
+        _camera.position.x = (float)(_gameData.getPlayerPosition().getX());
+        _camera.position.y = (float)(_gameData.getPlayerPosition().getY());
         _camera.update();
+        _batch.setProjectionMatrix(_camera.combined);
+        _batch.begin();
+        for (Entity e : _world.values())
+        {
+
+            View view = e.get(View.class);
+            Body body = e.get(Body.class);
+            Position pos = e.get(Position.class);
+            if (view.getImageFilePath().contains("ball"))
+            {
+                System.out.println("true? " + _assetManager.isLoaded(view.getImageFilePath()));
+                System.out.println("progress: " + _assetManager.getProgress());
+            }
+            if (body != null && pos != null && view != null && _assetManager.isLoaded(view.getImageFilePath()))
```

Comment:
this is very ugly. is it not inefficient to call isLoaded here? i suppose get returns null if the asset is not found anyways. just assign the return value above and check for null instead.